lib/elixir/lib/stream.ex

```diff
@@ -423,6 +423,10 @@ defmodule Stream do
   This operation will block the caller by the given interval
   every time a new item is streamed.
 
+  **Do not use this function** to generate number sequences. If
+  blocking the caller process is not necesary, use instead
+  `Stream.iterate(0, & &1 + 1)`.
```

Comment:
That's because you are using IEx so the anonymous function is being evaluated. Never benchmark in the shell:

```elixir
interval = Stream.interval(0)
IO.inspect :timer.tc(Enum, :take, [interval, 1000])

iterate = Stream.iterate(0, &(&1 + 1))
IO.inspect :timer.tc(Enum, :take, [interval, 1000])
```

Gives me a slightly faster iterate.